lib/datadog/statsd.rb

```diff
@@ -425,14 +425,27 @@ def flush_buffer()
       @buffer = Array.new
     end
 
+    def connect_to_socket(host, port)
+      socket = UDPSocket.new
+      socket.connect(host, port)
+      socket
+    rescue => boom
```

Comment:
`connect` shouldn't fail unless the host is unresolvable, and in this case we're probably fine with raising the exception (since it will be unabled to send anything anyway).
I'm not sure what would be the status of the socket if `connect` fails, most likely it would require using `send(message, 0, host, port)`, and thus would fail afterwards. So I think failing there is better than failing after.
Could you remove the error handling? (unless you think it's needed?)